crates/driver/src/domain/mempools.rs

```diff
@@ -34,6 +35,37 @@ impl Mempools {
             .boxed()
         })));
     }
+
+    /// Get gas price that is used for executing the settlement.
+    /// Since there are several mempools with different gas price estimators, we
+    /// will get them all and then pick the highest one.
+    pub async fn gas_price(&self, settlement: &Settlement) -> anyhow::Result<eth::GasPrice> {
+        let mut gas_prices = Vec::new();
+        for mempool in &self.0 {
+            let gas_price_estimator = mempool.gas_price_estimator(settlement);
+            let gas_price = gas_price_estimator.estimate().await?;
+            gas_prices.push(gas_price);
+        }
+
+        gas_prices.sort_unstable_by(|a, b| {
+            match a
+                .max_priority_fee_per_gas
+                .total_cmp(&b.max_priority_fee_per_gas)
+            {
+                std::cmp::Ordering::Equal => a.max_fee_per_gas.total_cmp(&b.max_fee_per_gas),
+                ordering => ordering,
+            }
+        });
+
+        gas_prices
+            .last()
+            .map(|gas_price| eth::GasPrice {
+                max: eth::U256::from_f64_lossy(gas_price.max_fee_per_gas).into(),
+                tip: eth::U256::from_f64_lossy(gas_price.max_priority_fee_per_gas).into(),
+                base: eth::U256::from_f64_lossy(gas_price.base_fee_per_gas).into(),
+            })
```

Comment:
Here we are doing conversion from one model to another in the domain. Should the gas estimator be wrapped in something that already returns the GasPrice type that is expected by the domain? cc @nlordell